### ISSUE TITLE
Make encryption compatible to other applications (e.g. Thunderbird with Enigmail)

### ISSUE THREAD

**Issue by mario98** [r10s/deltachat-core#27]
I was thrilled to learn that Delta Chat has now incorporated end-to-end encryption via. PGP, but I am a little bit disappointed how it is implemented. I can't get it to work with my existing software. I hope it's a bug and not by design.
I have already set up encryption prior to Delta Chat with Thunderbird and Enigmail. Now I wanted to set it up such that I can use my existing keys. I want to continue to use Thunderbird to read and even send Delta Chat messages.
So I tried to import my key generated with Enigmail into Delta Chat but this didn't work. I placed the .asc files (exported by Enigmail) in my Android download folder, even renamed them to "public-key-default" and "private-key-default", but Delta Chat said "no keys found" when using the (very limited) import feature in the advanced setting.
Also I am missing an option to import public keys from other users, for example, send as attachment by mail. (That's how key exchange works with Enigmail)
Delta Chat had the huge advantage that you could read messages by any mail application on desktop, web, other phones, etc. and you are not bound to a specific app. That's why I and my colleagues decided to use Delta Chat (besides from the fact that end-to-end encryption was promised for a future release). However with this new encryption feature which only works "in the background" with very little user configuration possible, this huge advantage seems to vanish.

Android L / Delta Chat 0.9.0
___
TL;DR: Please let us freely import, export and modify private and public keys. Thanks.

**Comment by r10s** [comment]
Hi mario98,

**all this is by design.**

First of all, the encryption format and the keys are compatible to other applications. Only the key transfer has changed.

Delta Chat wants to be an app for the normal folk, **not for specialists.** The old approach of key transfer has failed to see wide adoption outside of specialist communities, in large part because of difficulties with user experience and certification models.

The new Autocrypt approach lets users, that have _never_ encrypted a single mail before, encrypt their Mails by just _using_ Delta Chat. In fact, the user may not even know that he uses encryption.

Keep in mind, Delta Chat wants to be an alternative to WhatsApp an other messengers. Delta Chat wont reach this aim when it starts asking the user questions about keys and encryption he cannot answer. The alternative, not to encrypt, is no alternative but the current status.

But there are also some **good news for specialists:** You can **import** your existing private key at Settings / Menu button / Advanced settings / Manage private keys:

If you do so, recipients using Delta Chat will encrypt the mails in a way that they can be read by Delta Chat as well as your other clients Enigmail.
Moreover, Delta Chat can read mails from clients that are using the keys from a keyserver then.

And there are more good news: The Delta Chat encryption approach is not only a Delta Chat idea. In fact, there is a new group that wants to bring this new encryption approach to other clients, eg. K9, Enigmail and others, see https://autocrypt.readthedocs.io

![screenshot_20170603-003132](https://cloud.githubusercontent.com/assets/9800740/26747143/27183980-47f4-11e7-848c-230440ede870.png)

**Comment by mario98** [comment]
Thank you for your detailed answer.

However, as stated above, the import does not work for me. I tried to import my keys generated with Enigmail the way you described it, but it keeps saying "0 keys found". Enigmail generated normal asc files, aligned with the standard (as far as I can tell from looking at the files with a text editor) It seems that only specially formatted .asc file, as generated by Delta Chat, can be imported.

**Comment by r10s** [comment]
> It seems that only specially formatted .asc file, as generated by Delta Chat, can be imported.

This is not true. 

I use my old keys, Delta Chat follows the OpenPGP standard and does not expect a special format that is only created for Delta Chat. However, there are _lots_ of different keys algorithms out there - so it may be that a _concrete_ key does not work. Anyone is welcome to improve the support here.

BTW: The error message "0 keys found" may also come from the fact, the key file itself is not found. Are you sure, you put the file in the right directory? If in doubt, first do an key export and then search for the file private-key-default.asc in the downloads directory. Replace exactly this file with the private key and do an import then.

**Comment by r10s** [comment]
Moreover, make sure, you really add the *private* not the public key.

**Comment by r10s** [comment]
The upcoming version 0.9.1 will show more detailed errors on key import, I've just improved this part, see https://github.com/r10s/deltachat-core/commit/5ff33100f4512d97909385bd851d33eb6804c11a

**Comment by r10s** [comment]
With a very good chance, the problem is that you key is encrypted and/or uses an password. Such keys are not supported by Delta Chat. Please remove the encryption and the password and try the import again.

**Comment by mario98** [comment]
Indeed, the issue is a password. Enigmail is telling me that a password is strongly recommended and even does not allow me to remove it. (It may be a bug in Enigmail though) Anyway, thanks for helping to find the root of this issue. If you can add password support in a future release, I would appreciate this.

**Comment by r10s** [comment]
Yes, maybe in future releases.
